### PR TITLE
Format constants and static variables

### PR DESCRIPTION
**Pull request by marcusklaas**
Closes https://github.com/nrc/rustfmt/issues/431 and https://github.com/nrc/rustfmt/issues/465.

### CODE REVIEW

File: src/visitor.rs [hunk 1]
```diff
@@ -31,22 +32,8 @@ pub struct FmtVisitor<'a> {
...
-        let rewrite = ex.rewrite(&self.get_context(),
-                                 self.config.max_width - self.block_indent.width(),
-                                 self.block_indent);
-
-        if let Some(new_str) = rewrite {
-            self.buffer.push_str(&new_str);
-            self.last_pos = ex.span.hi;
-        }
+    fn visit_expr(&mut self, _: &'v ast::Expr) {
+        unreachable!()
```
Comment (nrc) [inline src/visitor.rs]: \o/

File: src/visitor.rs [hunk 2]
```diff
@@ -190,6 +179,7 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
             ast::Item_::ItemUse(ref vp) => {
                 self.format_import(item.vis, vp, item.span);
             }
+            // TODO(#78): format traits and impl definitions.
```
Comment (nrc) [inline src/visitor.rs]: nit: here and below, can we use FIXME rather than TODO for long-term things to be fixed?

File: src/visitor.rs [hunk 3]
```diff
@@ -225,7 +215,36 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
...
+                let rewrite = rewrite_static("static",
+                                             item.ident,
+                                             ty,
+                                             mutability,
+                                             expr,
+                                             &self.get_context());
+                if let Some(s) = rewrite {
+                    self.buffer.push_str(&s);
+                    self.last_pos = item.span.hi;
+                }
```
Comment (nrc) [inline src/visitor.rs]: Can we factor this little idiom into a function or macro?

**Comment by nrc** [comment]
r+ with nits addressed

File: src/visitor.rs (hunk 3 above)
Comment (marcusklaas) [inline src/visitor.rs]: That'd be nice. I'll try.

**Pull request merged**